src/sql/src/plan/statement/dml.rs

```diff
@@ -413,6 +428,53 @@ pub fn plan_explain_plan(
     }))
 }
 
+pub fn plan_explain_schema(
+    scx: &StatementContext,
+    explain_schema: ExplainSchemaStatement<Aug>,
+) -> Result<Plan, PlanError> {
+    let ExplainSchemaStatement {
+        schema_for,
+        format,
+        mut statement,
+    } = explain_schema;
+    let format = match format {
+        mz_sql_parser::ast::ExplainFormat::Text => ExplainFormat::Text,
+        mz_sql_parser::ast::ExplainFormat::Json => ExplainFormat::Json,
+        mz_sql_parser::ast::ExplainFormat::Dot => ExplainFormat::Dot,
+    };
+    crate::pure::add_materialize_comments(scx.catalog, &mut statement)?;
+
+    if let Plan::CreateSink(CreateSinkPlan { sink, .. }) = ddl::plan_create_sink(scx, statement)? {
```

Comment:
nit: you can avoid making all the logic nested if instead of:

```
if let Pattern(x) = something {
    // logic
} else {
    unreachable!()
}
```

you write:

```
let Pattern(x) = something else {
    unreachable!();
}
// logic ...
```

This also keeps the `unreachable` statement close to the check it's related to.

Reply:
In this case though there was one particular branch i.e. a create sink plan was valid while there would have been multiple cases/plans which would have been the 'something else' for unreachable. I updated it to match statements though, should be slightly easier to grok?